pkg/resources/kafka/kafka.go

```diff
@@ -905,6 +911,30 @@ func (r *Reconciler) createExternalListenerStatuses(log logr.Logger) (map[string
 	return extListenerStatuses, nil
 }
 
+func (r *Reconciler) getK8sAssignedNodeport(log logr.Logger, eListenerName string, brokerId int32) (int32, error) {
+	log.Info("determining automatically assigned nodeport",
+		"brokerId", brokerId, "listenerName", eListenerName)
+	nodePortSvc := &corev1.Service{}
+	err := r.Client.Get(context.Background(),
+		types.NamespacedName{Name: fmt.Sprintf("%s-%d-%s", r.KafkaCluster.GetName(), brokerId, eListenerName),
+			Namespace: r.KafkaCluster.GetNamespace()}, nodePortSvc)
+	if err != nil {
+		return 0, errors.WrapWithDetails(err, "could not get nodeport service",
+			"brokerId", brokerId, "listenerName", eListenerName)
+	}
+	log.Info("nodeport service successfully found",
+		"brokerId", brokerId, "listenerName", eListenerName)
+
+	var nodePort int32
+	for _, port := range nodePortSvc.Spec.Ports {
```

Comment:
Is the port number assigned to `port.NodePort` instantaneously by K8s or there might be some delay thus we need to wait here until the port is set in case `port.NodePort` is 0?

Reply:
Accourding to the documentation it happens instantaneously. It is like assigning the clusterIP to the clusterIP types service. I check the internet regarding this error but I found nothing.